### ISSUE TITLE
pyemma.msm.analysis.timescales returns two infinite implied time scales for a connected transition matrix

### ISSUE THREAD

**Issue by fabian-paul** [markovmodel/PyEMMA#30]
pyemma.msm.analysis.timescales has some in-built functionality to decide whether an implied time scale is infinite. If think its tolerance is too low. Consider the example below
```python
import numpy as np
import pyemma
eps = 1E-6
T=np.zeros((2,2)); T[1,0] = eps; T[0,1] = eps; np.fill_diagonal(T,1.0-T.sum(axis=1))
l,_=np.linalg.eig(T)
print -1.0/np.log(l) # [ -inf  499999.49998546]
print pyemma.msm.analysis.timescales(T,tau=1.0) # [ inf  inf]
```
Is there any reason why 499999.49998546 is replaced by inf? For a connected T-matrix, I would expect that there is only one infinite implied time scale.

**Comment by franknoe** [comment]
I agree with Fab. What is the reasoning of doing 
    ind_abs_one=isclose(np.abs(evals), 1.0)
    ts[ind_abs_one]=np.inf
in the conversion of eigenvalues to timescales? It just appears to introduce an unnecessary error. If the eigenvalue is smaller than 1.0, the timescales is finite. 
Perhaps there are reasonable bounds for the timescales when they come from a transition matrix estimated from MD data, but for model systems timescales could be abitrarily large. Since the timescales methods knows nothing about the origin of the transition matrix, I would just set values >= 1 to infinity.

**Comment by trendelkampschroer** [comment]
Please try out the following snippet of code

x = 1.0 - tol
-1.0/np.log(x)

with tol = 1e-14, 0.5*e-16

The test is just there to catch the error-message you should see.

 Unfortunately the default tolerance in the isclose function is too low (1e-8 or so). I already submitted a fix, which did not pass because of some strange test behavior, occuring only during automatic build.

**Comment by franknoe** [comment]
thanks!

what about catching this error message and then setting the timescale to 
inf only in response?

Am 12/03/15 um 17:14 schrieb Benjamin Trendelkamp-Schroer:
>
> Please try out the following snippet of code
>
> x = 1.0 - tol
> -1.0/np.log(x)
>
> with tol = 1e-14, 0.5*e-16
>
> The test is just there to catch the error-message you should see.
>
> Unfortunately the default tolerance in the isclose function is too low 
> (1e-8 or so). I already submitted a fix, which did not pass because of 
> some strange test behavior, occuring only during automatic build.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/markovmodel/PyEMMA/issues/30#issuecomment-78513328>.
>


-- 

----------------------------------------------
Prof. Dr. Frank Noe
Head of Computational Molecular Biology group
Freie Universitaet Berlin

Phone: (+49) (0)30 838 75354
Web:   research.franknoe.de
Mail:  Arnimallee 6, 14195 Berlin, Germany
----------------------------------------------

**Comment by trendelkampschroer** [comment]
In principle I want to compute all timescales at once. 

I have to split the computation. If I just catch the error message I will never know which eigenvalues lead to the error and I could not distinguish between the finite and the infinite timescales.

**Comment by franknoe** [comment]
True, but that would only occur for the few ones very close to 1, so 
it's not a very big extra effort.

Am 12/03/15 um 19:04 schrieb Benjamin Trendelkamp-Schroer:
>
> In principle I want to compute all timescales at once.
>
> I have to split the computation. If I just catch the error message I 
> will never know which eigenvalues lead to the error and I could not 
> distinguish between the finite and the infinite timescales.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/markovmodel/PyEMMA/issues/30#issuecomment-78549779>.
>


-- 

----------------------------------------------
Prof. Dr. Frank Noe
Head of Computational Molecular Biology group
Freie Universitaet Berlin

Phone: (+49) (0)30 838 75354
Web:   research.franknoe.de
Mail:  Arnimallee 6, 14195 Berlin, Germany
----------------------------------------------

**Comment by franknoe** [comment]
or: only compute them iterative as a fall-back solution, i.e. if that 
error occurs for the vector operation.

Am 12/03/15 um 19:04 schrieb Benjamin Trendelkamp-Schroer:
>
> In principle I want to compute all timescales at once.
>
> I have to split the computation. If I just catch the error message I 
> will never know which eigenvalues lead to the error and I could not 
> distinguish between the finite and the infinite timescales.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/markovmodel/PyEMMA/issues/30#issuecomment-78549779>.
>


-- 

----------------------------------------------
Prof. Dr. Frank Noe
Head of Computational Molecular Biology group
Freie Universitaet Berlin

Phone: (+49) (0)30 838 75354
Web:   research.franknoe.de
Mail:  Arnimallee 6, 14195 Berlin, Germany
----------------------------------------------

**Comment by franknoe** [comment]
Fixed by PR #298
